codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/ModelAuthSchemeKnowledgeIndex.java

```diff
@@ -105,17 +157,126 @@ private Map<List<String>, List<AuthTrait>> operationsToAuthOptions() {
                                       Map.Entry::getKey, (a, b) -> b,
                                       LinkedHashMap::new));
 
-        List<AuthTrait> serviceDefaults = serviceDefaultAuthOption();
+        List<AuthTrait> serviceDefaults = serviceDefaultAuthTrait();
 
-        // Handle operations with defaults
+        // Get the list of operations that share the same auth schemes as the system defaults and remove it from the result. We
+        // will take care of all of these in the fallback `default` case.
         List<String> operationsWithDefaults = authSchemesToOperations.remove(serviceDefaults);
         if (operationsWithDefaults != null) {
             operationsToAuthTrait.remove(operationsWithDefaults);
         }
-        operationsToAuthTrait.put(Collections.emptyList(), serviceDefaults);
+
         return operationsToAuthTrait;
     }
 
+    /**
+     * Determines if an operation should be included based on whether it has auth traits.
```

Comment:
minor: It's not clear what the operation should be included in. Could probably just remove that part.